includes/controllers/pages/class-submit-listing.php

```diff
@@ -136,17 +136,23 @@ public function dispatch() {
         }
 
         if ( ! $this->editing && 'auto-draft' !== get_post_status( $this->listing->get_id() ) ) {
-            $possible_payment = WPBDP_Payment::objects()->filter(
-                array(
-					'listing_id'   => $this->listing->get_id(),
-					'payment_type' => 'initial',
-					'status'       => 'pending',
-                )
-            )->get();
-
-            if ( $possible_payment ) {
-                return $this->_redirect( $possible_payment->get_checkout_url() );
-            }
+			$plan_id = absint( wpbdp_get_var( array( 'param' => 'listing_plan', 'default' => 0 ), 'post' ) );
+			$plan    = wpbdp_get_fee_plan( $plan_id );
+            if ( $plan && $plan->enabled ) {
+				$plan = $this->get_plan_for_listing();
```

Comment:
It looks like this line should be removed? This doesn't make sense to me.

Reply:
Agreed